kevlar/call.py

```diff
@@ -161,79 +170,128 @@ def call_snv(target, query, offset, length, ksize):
-    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=window,
-                         RW=refrwindow, IK=str(len(query.ikmers)))]
+    return refr, alt, refrwindow, altwindow
 
 
-def call_insertion(target, query, offset, ksize, leftmatch, indellength):
+def insertion_allele(target, query, offset, ksize, leftmatch, indellength):
     minpos = leftmatch - ksize + 1
     maxpos = leftmatch + ksize + indellength - 1
-    window = query.sequence[minpos:maxpos]
+    altwindow = query.sequence[minpos:maxpos]
     minpos += offset
     maxpos += offset - indellength
     refrwindow = target.sequence[minpos:maxpos]
 
-    insertion = query.sequence[leftmatch-1:leftmatch+indellength]
-    refr = insertion[0]
-    assert len(insertion) == indellength + 1
-    localcoord = offset + leftmatch
+    alt = query.sequence[leftmatch-1:leftmatch+indellength]
+    refr = alt[0]
+    return refr, alt, refrwindow, altwindow
+
+
+def call_deletion(target, query, offset, ksize, leftmatch, indellength):
+    if offset < 0:
+        offset *= -1
+        targetshort = True
+        alt, refr, altwindow, refrwindow = insertion_allele(
+            query, target, offset, ksize, leftmatch, indellength
+        )
+    else:
+        targetshort = False
+        refr, alt, refrwindow, altwindow = deletion_allele(
+            target, query, offset, ksize, leftmatch, indellength
+        )
+    # This assertion is no longer valid when query is longer than target
+    # assert len(refr) == indellength + 1
+    localcoord = leftmatch
+    if not targetshort:
+        localcoord += offset
+    seqid, globalcoord = local_to_global(localcoord, target.name)
+    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=altwindow,
+                         RW=refrwindow, IK=str(len(query.ikmers)))]
+
+
+def call_insertion(target, query, offset, ksize, leftmatch, indellength):
+    if offset < 0:
+        offset *= -1
+        targetshort = True
+        alt, refr, altwindow, refrwindow = deletion_allele(
+            query, target, offset, ksize, leftmatch, indellength
+        )
+    else:
+        targetshort = False
+        refr, alt, refrwindow, altwindow = insertion_allele(
+            target, query, offset, ksize, leftmatch, indellength
+        )
+
+    assert len(alt) == indellength + 1
+    localcoord = leftmatch
+    if not targetshort:
+        localcoord += offset
     seqid, globalcoord = local_to_global(localcoord, target.name)
-    return [VariantIndel(seqid, globalcoord - 1, refr, insertion, VW=window,
+    return [VariantIndel(seqid, globalcoord - 1, refr, alt, VW=altwindow,
                          RW=refrwindow, IK=str(len(query.ikmers)))]
 
 
 def make_call(target, query, cigar, ksize):
-    snvmatch = re.search('^(\d+)D(\d+)M(\d+)D$', cigar)
-    snvmatch2 = re.search('^(\d+)D(\d+)M(\d+)D(\d+)M$', cigar)
+    snvmatch = re.search('^(\d+)([DI])(\d+)M(\d+)[DI]$', cigar)
+    snvmatch2 = re.search('^(\d+)([DI])(\d+)M(\d+)[DI](\d+)M$', cigar)
```

Comment:
This is the critical change to the PR with respect to SNVs. Most other changes in this file were to compensate for this change.